plaidml/bridge/openvino/ops/embedding_segments_sum.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+
+namespace {
+
+template <typename T>
+std::vector<T> cast_constant_operand(size_t operand_idx, ngraph::Node* layer) {
+  auto* ngraph_const = ngraph::as_type<ngraph::op::Constant>(layer->get_input_node_ptr(operand_idx));
+  if (ngraph_const) {
+    return ngraph_const->cast_vector<T>();
+  } else {
+    THROW_IE_EXCEPTION << "Dynamic slicing not currently supported by PlaidML plugin; all of indices, segment_ids, "
+                          "num_segments and default index must be Constants.";
+  }
+}
+
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+void registerEmbeddingSegmentsSum() {
+  registerOp("EmbeddingSegmentsSum", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<ngraph::opset4::EmbeddingSegmentsSum>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() >= 4);
+    IE_ASSERT(ctx.operands.size() <= 6);
+    auto I = ctx.operands.at(0);
+    auto indices = ctx.operands.at(1);
+    IE_ASSERT(indices.rank() == 1);
+    auto segment_ids = ctx.operands.at(2);
+    auto num_segments = cast_constant_operand<int32_t>(3, layer)[0];
+    int32_t default_index = -1;
```

Comment:
nit: maybe use a boolean like  with_weights is better than setting  -1 as a default value. generally we should avoid using magic numbers in the code.. An alternaive would be using a static constexpr.

Reply:
OK then we have to use 2 variables:
```c++
bool with_default_index
int32_t default_index
```
Is this better?

Reply:
Yes, this would be good.